pxtblocks/blocklyloader.ts

```diff
@@ -2269,6 +2273,268 @@ namespace pxt.blocks {
+                    y = xy.y + (Blockly as any).SNAP_RADIUS * 2;
+                }
+                let xml = goog.dom.createDom('xml');
+                let block = goog.dom.createDom('block');
+                block.setAttribute('type', 'procedures_defnoreturn');
+                block.setAttribute('x', String(x));
+                block.setAttribute('y', String(y));
+                let field = goog.dom.createDom('field');
+                field.setAttribute('name', 'NAME');
+                field.appendChild(document.createTextNode(name));
+                block.appendChild(field);
+                xml.appendChild(block);
+                let newBlockIds = Blockly.Xml.domToWorkspace(xml, workspace);
+                // Close flyout and highlight block
+                (workspace as any).toolbox_.clearSelection();
+                let newBlock = workspace.getBlockById(newBlockIds[0]);
+                newBlock.select();
+            }
+
+            workspace.registerButtonCallback('CREATE_FUNCTION', function(button) {
+                let promptAndCheckWithAlert = (defaultName: string) => {
+                    Blockly.prompt(newFunctionTitle, defaultName, function(newFunc) {
+                        // Merge runs of whitespace.  Strip leading and trailing whitespace.
+                        // Beyond this, all names are legal.
+                        if (newFunc) {
+                            newFunc = newFunc.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
+                            if (newFunc == newFunction) {
+                                // Ok, not ALL names are legal...
+                                newFunc = null;
+                            }
+                        }
+                        if (newFunc) {
+                            if (workspace.variableIndexOf(newFunc) != -1) {
+                                Blockly.alert((Blockly as any).Msg.VARIABLE_ALREADY_EXISTS.replace('%1',
+                                    newFunc.toLowerCase()),
+                                    function() {
+                                        promptAndCheckWithAlert(newFunc);  // Recurse
+                                    });
+                            }
+                            else if (!Blockly.Procedures.isLegalName_(newFunc, workspace)) {
+                                Blockly.alert((Blockly as any).Msg.PROCEDURE_ALREADY_EXISTS.replace('%1',
+                                    newFunc.toLowerCase()),
+                                    function() {
+                                        promptAndCheckWithAlert(newFunc);  // Recurse
+                                    });
+                            }
+                            else {
+                                createFunction(newFunc);
+                            }
+                        }
+                    });
+                };
+                promptAndCheckWithAlert('');
```

Comment:
Okay. I chose to default to doSomething in monaco? should I update both to foo? thoughts on which is a better default?